### PR TITLE
Convert CorePrefsFragment.java into kotlin

### PR DESCRIPTION
**Pull request by MohitMaliFtechiz**
Fixes: #3041 
Parent Issue: #1826 

Converted CorePrefsFragment.java into kotlin

### CODE REVIEW

**Review by gouri-panda — COMMENTED** [review]
@MohitMaliDeveloper It keeps failing on API level 21.

File: core/src/main/java/org/kiwix/kiwixmobile/core/settings/CorePrefsFragment.kt [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+    )?.let {
+      var languageCodeList = LanguageUtils(requireActivity()).keys
+      languageCodeList = listOf(Locale.ROOT.language) + languageCodeList
+      sharedPreferenceUtil?.let { sharedPreferenceUtil ->
+        val selectedLang = selectedLanguage(languageCodeList, sharedPreferenceUtil.prefLanguage)
+        it.entries = languageDisplayValues(languageCodeList)
+        it.entryValues = languageCodeList.toTypedArray()
+        it.setDefaultValue(selectedLang)
+        it.value = selectedLang
+        it.title =
```
Comment (gouri-panda) [inline core/src/main/java/org/kiwix/kiwixmobile/core/settings/CorePrefsFragment.kt:140]: Maybe we should replace the lambda parameter `it` with `languagePref` for the sake of readabilty.

File: core/src/main/java/org/kiwix/kiwixmobile/core/settings/CorePrefsFragment.kt [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+            }
+            true
+          }
+      }
+    }
+  }
+
+  private fun restartActivity() {
+    val activity = activity as CoreMainActivity?
+    activity?.let {
```
Comment (gouri-panda) [inline core/src/main/java/org/kiwix/kiwixmobile/core/settings/CorePrefsFragment.kt:162]: We can make it in one line.

**Comment by MohitMaliFtechiz** [comment]
> 

hi @gouri-panda 
i have re-run the job and it pass now.

File: core/src/main/java/org/kiwix/kiwixmobile/core/settings/CorePrefsFragment.kt [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+  var sharedPreferenceUtil: SharedPreferenceUtil? = null
+
+  @JvmField @Inject
+  var storageCalculator: StorageCalculator? = null
+
+  @JvmField @Inject
+  var nightModeConfig: NightModeConfig? = null
+
+  @JvmField @Inject
+  var alertDialogShower: DialogShower? = null
```
[the remark this answers is not in the document]
Comment (gouri-panda) [reply core/src/main/java/org/kiwix/kiwixmobile/core/settings/CorePrefsFragment.kt:77]: @MohitMaliDeveloper What about protected fields?

[the remark this answers is not in the document]
Comment (gouri-panda) [reply core/src/main/java/org/kiwix/kiwixmobile/core/settings/CorePrefsFragment.kt:77]: sorry!, what I meant was does dagger support injection into protected fields? I have never done that.

[the remark this answers is not in the document]
Comment (MohitMaliFtechiz) [reply core/src/main/java/org/kiwix/kiwixmobile/core/settings/CorePrefsFragment.kt:77]: @gouri-panda ,

yes dagger support injection into protected filed. it only not supported in private fields.

[the remark this answers is not in the document]
Comment (gouri-panda) [reply core/src/main/java/org/kiwix/kiwixmobile/core/settings/CorePrefsFragment.kt:77]: ok, then we should use protected.

[the remark this answers is not in the document]
Comment (MohitMaliFtechiz) [reply core/src/main/java/org/kiwix/kiwixmobile/core/settings/CorePrefsFragment.kt]: i have done the changes

File: core/src/main/java/org/kiwix/kiwixmobile/core/settings/CorePrefsFragment.kt [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+        requireActivity().layoutInflater,
+        it
+      )
+    }
+  }
+
+  private fun setupZoom() {
+    val textZoom = findPreference<Preference>(INTERNAL_TEXT_ZOOM)
+    textZoom?.onPreferenceChangeListener =
+      Preference.OnPreferenceChangeListener { _: Preference?, newValue: Any ->
```
[the remark this answers is not in the document]
Comment (MohitMaliFtechiz) [reply core/src/main/java/org/kiwix/kiwixmobile/core/settings/CorePrefsFragment.kt]: done

File: core/src/main/java/org/kiwix/kiwixmobile/core/settings/CorePrefsFragment.kt (hunk 1 above)
Comment (MohitMaliFtechiz) [reply core/src/main/java/org/kiwix/kiwixmobile/core/settings/CorePrefsFragment.kt]: done

File: core/src/main/java/org/kiwix/kiwixmobile/core/settings/CorePrefsFragment.kt (hunk 2 above)
Comment (MohitMaliFtechiz) [reply core/src/main/java/org/kiwix/kiwixmobile/core/settings/CorePrefsFragment.kt]: done

File: core/src/main/java/org/kiwix/kiwixmobile/core/settings/CorePrefsFragment.kt [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+  @Suppress("NestedBlockDepth")
+  private fun onStorageDeviceSelected(storageDevice: StorageDevice) {
+    sharedPreferenceUtil?.let { sharedPreferenceUtil ->
+      findPreference<Preference>(SharedPreferenceUtil.PREF_STORAGE)?.summary =
+        storageCalculator?.calculateAvailableSpace(storageDevice.file)
+      if (storageDevice.isInternal) {
+        sharedPreferenceUtil.putPrefStorage(
+          sharedPreferenceUtil.getPublicDirectoryPath(storageDevice.name)
+        )
+        findPreference<Preference>(SharedPreferenceUtil.PREF_STORAGE)!!.title =
```
Comment (gouri-panda) [inline core/src/main/java/org/kiwix/kiwixmobile/core/settings/CorePrefsFragment.kt:306]: Remove the double bang operator. We don't encourage double bang in this project unless we wanna crash if something goes really wrong. That's how we gonna get the crash reports :)

**Review by gouri-panda — COMMENTED** [review]
@MohitMaliDeveloper Just some small changes then we are good to go! I have tested both kiwix app and custom app everything looks fine.

File: core/src/main/java/org/kiwix/kiwixmobile/core/settings/CorePrefsFragment.kt (hunk 5 above)
Comment (MohitMaliFtechiz) [reply core/src/main/java/org/kiwix/kiwixmobile/core/settings/CorePrefsFragment.kt]: hi @gouri-panda  

done

**Review by gouri-panda — APPROVED** [review]

**Pull request merged**